src/bytes.js

```diff
@@ -0,0 +1,1455 @@
+                val1 = object.v[i];
+            }
+            val2 = suff.v[j];
+            if (val1 != val2) {
+                return Sk.builtin.bool.$false;
+            }
+        }
+        return Sk.builtin.bool.$true;
+    };
+
+    if (suffix instanceof Sk.builtin.tuple) {
+        for (iter = Sk.abstr.iter(suffix), item = iter.tp$iternext();
+            item !== undefined;
+            item = iter.tp$iternext()) {
+            if (!(item instanceof Sk.builtin.bytes)) {
+                throw new Sk.builtin.TypeError("a bytes-like object is required, not '" + Sk.abstr.typeName(item) + "'");
+            }
+            len = item.v.byteLength;
+            if ((endidx - startidx) >= len) {
+                if (endidx < 0) {
+                    if (negend(endidx, self, item) === Sk.builtin.bool.$true) {
+                        return Sk.builtin.bool.$true;
+                    }
+                }
+                index = new Sk.builtin.slice(endidx - len, endidx);
+                val = self.mp$subscript(index);
+                if (val.ob$eq(item) === Sk.builtin.bool.true$) {
+                    return Sk.builtin.bool.true$;
+                }
+            }
+        }
+        return Sk.builtin.bool.false$;
+    } else {
+        len = suffix.v.byteLength;
+        if ((endidx - startidx) >= len) {
+            if (endidx < 0) {
+                return (negend(endidx, self, suffix));
+            }
+            index = new Sk.builtin.slice(endidx - len, endidx);
+            val = self.mp$subscript(index);
+            if (val.ob$eq(suffix) === Sk.builtin.bool.true$) {
+                return Sk.builtin.bool.true$;
+            }
+        }
+        return Sk.builtin.bool.false$;
+    }
+});
+
+Sk.builtin.bytes.prototype.find_item_ = function (sub, start, end) {
+    var i;
+    var len;
+    var val;
+
+    let { startidx, endidx } = indices(this, start, end);
+
+    if (sub instanceof Sk.builtin.int_) {
+        for (i = startidx; i < endidx; i++) {
+            if (this.v[i] === sub.v) {
+                return new Sk.builtin.int_(i);
+            }
+        }
+        return new Sk.builtin.int_(-1);
+    } else if (sub instanceof Sk.builtin.bytes) {
+        len = sub.v.byteLength;
+        while (startidx + len <= endidx) {
+            i = new Sk.builtin.slice(startidx, startidx + len);
+            val = this.mp$subscript(i);
+            if (val.ob$eq(sub) === Sk.builtin.bool.true$) {
+                return new Sk.builtin.int_(startidx);
+            }
+            startidx++;
+        }
+        return new Sk.builtin.int_(-1);
+    } else {
+        throw new Sk.builtin.TypeError("argument should be integer or bytes-like object, not '" + Sk.abstr.typeName(sub) + "'");
+    }
+};
+
+Sk.builtin.bytes.prototype["find"] = new Sk.builtin.func(function (self, sub, start, end) {
+    Sk.builtin.pyCheckArgsLen("find", arguments.length - 1, 1, 3);
+
+    return Sk.builtin.bytes.prototype.find_item_.call(self, sub, start, end);
```

Comment:
suggestion
change this to `Sk.builtin.bytes.prototype.find$item`
and instead of doing `Sk.builtin.bytes.prototype.find_item_.call` do:
```javascript
self.find$item(sub, start, end)
```
or use it as a standalone function in this module rather than on the `prototype` of bytes.

(i only suggest `$` because in theory a trailing underscore could be overridden by a subclass)